overlord/hookstate/ctlcmd/health.go

```diff
@@ -0,0 +1,147 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/overlord/healthstate"
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+var (
+	shortHealthHelp = i18n.G("Report on snap's health")
+	longHealthHelp  = i18n.G(`
+The set-health command is called from a snap to inform the system of the snap's
+overall health.
+
+It can be called from any hook, and even from the apps themselves, although a
+snap can provide a 'check-health' hook that is solely concerned with this; if it
+does, that hook is called periodically (with increased frequency while the snap
+is "unhealthy"). Any health regression will issue a warning to the user.
+
+Note the health is of the snap, not of the apps it contains; it’s up to the
+snap developer to determine how the health of the individual apps add up to
+the health of the snap as a whole.
+
+status can be one of
+
+- okay: the snap is healthy. This status takes no message and no code.
+
+- waiting: some resource (e.g. a device, network, or service) the snap needs
+  isn’t ready yet; the user just needs to wait.  The message must explain what
+  it’s waiting for.
```

Comment:
waiting: a resource needed by the snap (e.g. a device, network, or service)  is not ready and the user will need to wait. The message must explain which resource is being waited for.